include/pulsar/ProducerConfiguration.h

```diff
@@ -532,6 +532,18 @@ class PULSAR_PUBLIC ProducerConfiguration {
      */
     ProducerAccessMode getAccessMode() const;
 
+    /**
+     * Use this configuration to automatically create an initial subscription when creating a topic.
+     *
+     * If this field is not set, the initial subscription is not created.
+     */
+    ProducerConfiguration& setInitialSubscriptionName(const std::string& initialSubscriptionName);
+
+    /**
+     * Get initial subscription name.
+     */
+    const std::string& getInitialSubscriptionName() const;
+
```

Comment:
These methods should not be exposed to users, see https://github.com/apache/pulsar/pull/13355/files. We should access `ProducerConfigurationImpl::initialSubscriptionName`  directly.